packages/@aws-cdk/aws-kinesisfirehose/lib/delivery-stream.ts

```diff
@@ -0,0 +1,256 @@
+  /**
+   * The destinations that this delivery stream will deliver data to.
+   *
+   * Only a singleton array is supported at this time.
+   */
+  readonly destinations: IDestination[];
+
+  /**
+   * A name for the delivery stream.
+   *
+   * @default - a name is generated by CloudFormation.
+   */
+  readonly deliveryStreamName?: string;
+
+  /**
+   * The IAM role assumed by Kinesis Data Firehose to read from sources, invoke processors, and write to destinations.
+   *
+   * @default - a role will be created with default permissions.
+   */
+  readonly role?: iam.IRole;
+}
+
+/**
+ * A full specification of a delivery stream that can be used to import it fluently into the CDK application.
+ */
+export interface DeliveryStreamAttributes {
+  /**
+   * The ARN of the delivery stream.
+   *
+   * At least one of deliveryStreamArn and deliveryStreamName must be provided.
+   *
+   * @default - derived from `deliveryStreamName`.
+   */
+  readonly deliveryStreamArn?: string;
+
+  /**
+   * The name of the delivery stream
+   *
+   * At least one of deliveryStreamName and deliveryStreamArn  must be provided.
+   *
+   * @default - derived from `deliveryStreamArn`.
+   */
+  readonly deliveryStreamName?: string;
+
+
+  /**
+   * The IAM role associated with this delivery stream.
+   *
+   * Assumed by Kinesis Data Firehose to read from sources, invoke processors, and write to destinations.
+   *
+   * @default - the imported stream cannot be granted access to other resources as an `iam.IGrantable`.
+   */
+  readonly role?: iam.IRole;
+}
+
+/**
+ * Create a Kinesis Data Firehose delivery stream
+ *
+ * @resource AWS::KinesisFirehose::DeliveryStream
+ */
+export class DeliveryStream extends DeliveryStreamBase {
+  /**
+   * Import an existing delivery stream from its name.
+   */
+  static fromDeliveryStreamName(scope: Construct, id: string, deliveryStreamName: string): IDeliveryStream {
+    return this.fromDeliveryStreamAttributes(scope, id, { deliveryStreamName });
+  }
+
+  /**
+   * Import an existing delivery stream from its ARN.
+   */
+  static fromDeliveryStreamArn(scope: Construct, id: string, deliveryStreamArn: string): IDeliveryStream {
+    return this.fromDeliveryStreamAttributes(scope, id, { deliveryStreamArn });
+  }
+
+  /**
+   * Import an existing delivery stream from its attributes.
+   */
+  static fromDeliveryStreamAttributes(scope: Construct, id: string, attrs: DeliveryStreamAttributes): IDeliveryStream {
+    if (!attrs.deliveryStreamName && !attrs.deliveryStreamArn) {
+      throw new Error('Either deliveryStreamName or deliveryStreamArn must be provided in DeliveryStreamAttributes');
+    }
+    const deliveryStreamName = attrs.deliveryStreamName ?? cdk.Stack.of(scope).parseArn(attrs.deliveryStreamArn!).resourceName;
+    if (!deliveryStreamName) {
+      throw new Error(`Could not import delivery stream from malformatted ARN ${attrs.deliveryStreamArn}: could not determine resource name`);
```

Comment:
Use the service-specific term here:

```suggestion
      throw new Error(`No stream name found in ARN: '${attrs.deliveryStreamArn}'`);
```